tiledb/sm/rest/rest_profile.cc

```diff
@@ -178,6 +177,14 @@ std::string RestProfile::get_param(const std::string& param) const {
  * See issue [#727](https://github.com/nlohmann/json/issues/727) for details.
  */
 void RestProfile::save_to_file() {
+  // RestProfiles are immutable, so disallow overwrites.
+  if (saved_) {
+    throw RestProfileException(
+        "Failed to save \'" + name_ +
+        "\'; This profile has already been saved, "
```

Comment:
The "saved" nomenclature is at odds with the notion that you can only do it once. In other workflows saving overwrites whatever the previous state was.

Published? Committed? I'm not sure but either of these terms carries a little more weight towards "this thing is immutable once written".  And of course you could just say that too.

Reply:
`saved_` represents "this `Profile` has had the `save()` method invoked on it and has been therefore immutable written to disk". 
Such is the inline documentation for the variable. I suppose you could use another name; this is simply an internal flag to be flipped when the immutable write occurs.

Reply:
There is no "saved" anymore.